### PR TITLE
Use long value so that cache sizes greater than 2Gigabytes can be supported

### PR DESCRIPTION
**Pull request by bhargavms**
## Description
I want to be able to set cache sizes greater than 2 gigabytes which is the max value supported by the int primitive type, my approact to achieving this is to change the type used for maintaing cache size to long, so that cache with greater than 2 gigabyte can be used since these days phones are getting bulkier I think its fair to also let developers use more of the phone to improve User experience.

Issue Reference:
https://github.com/bumptech/glide/issues/2022

## Motivation and Context
Bigger cache sizes equals better user experiences!

### CODE REVIEW

**Comment by bhargavms** [comment]
CLA signed

**Comment by sjudd** [comment]
The changes you've made are only to the classes that hold data in memory. You want memory caches > 2GB?

**Comment by bhargavms** [comment]
@sjudd oh yes, forgot to change diskcache too, will update PR, yes user should have option to set whatever cache size he wants is my opinion.

**Comment by bhargavms** [comment]
CLA signed

**Comment by bhargavms** [comment]
@sjudd if you don't like using `long` for memory cache size, I can revert commit and squash, I'm not terribly passionate about that part.

File: third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java [hunk 1]
```diff
@@ -418,8 +418,20 @@ private Bitmap setPixels(GifFrame currentFrame, GifFrame previousFrame) {
     // Final location of blended pixels.
     final int[] dest = mainScratch;
 
-    // clear all pixels when meet first frame
+    // clear all pixels when meet first frame and drop prev image from last loop
```
Comment (sjudd) [inline third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java]: I think this means you need to rebase.

**Comment by sjudd** [comment]
I don't object to using longs for memory cache/bitmap pool. It's a little overkill now, but not problematic. LGTM. 

It may be another day or two before I merge this, I think there's going to be a dot release of 4.3 which I'd like to get out first.

File: third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java (hunk 1 above)
Comment (bhargavms) [reply third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java]: @sjudd I will close this PR and open another one by rebasing correctly

**Pull request closed without merging**